lib/rules/valid-api.js

```diff
@@ -68,7 +68,7 @@ function validatePropertyName(property, context) {
         });
     }
 
-    if (name.startsWith('data')) {
+    if (name.startsWith('data') && name !== 'data') {
```

Comment:
The reserved public property check for `dataset` will never be hit unless we move this check below that one.